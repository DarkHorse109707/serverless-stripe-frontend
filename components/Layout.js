import PropTypes from 'prop-types';
import Head from 'next/head';

const Layout = props => (
  <div>
    <Head>
      <title>Serverless Stripe Checkout Example</title>
      <script src="https://js.stripe.com/v3/" />
      <style>{`
         html {
           height: 100%;
           background: #F6F8FA;
         }
         #container {
           font-family: BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif;
           color: #4C555A;
           min-height: 100%;
           text-align: center;
           -webkit-font-smoothing: antialiased;
         }
         a {
           color: #00a3da;
         }
         a:hover {
           background-color: #fff2a8;
         }
      `}</style>
    </Head>
    <div id="container">
      {props.children}
    </div>
  </div>
);

Layout.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Layout;
