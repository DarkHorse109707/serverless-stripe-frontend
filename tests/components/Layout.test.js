import React from 'react';
import renderer from 'react-test-renderer';

import Layout from '../../components/Layout';

describe('Layout', () => {
  test('renders successfully', () => {
    const component = renderer.create(<Layout><p>Hello world</p></Layout>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
