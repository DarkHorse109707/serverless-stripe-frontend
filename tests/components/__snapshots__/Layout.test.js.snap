// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders successfully 1`] = `
<div>
  <div
    id="container"
  >
    <p>
      Hello world
    </p>
  </div>
</div>
`;
